src/snowflake/snowpark/functions.py

```diff
@@ -1083,14 +1153,44 @@ def soundex(e: ColumnOrName) -> Column:
 
 
 def trim(e: ColumnOrName, trim_string: Optional[ColumnOrName] = None) -> Column:
-    """Removes leading and trailing characters from a string."""
+    """Removes leading and trailing characters from a string. Per default only whitespace ' ' characters are removed.
+
+    Example::
+
+        >>> df = session.create_dataframe(['hello', ' world', '   !   '], schema=["a"])
+        >>> df.collect()
+        [Row(A='hello'), Row(A=' world'), Row(A='   !   ')]
+        >>> df.select(trim(col("a"))).collect()
+        [Row(TRIM("A")='hello'), Row(TRIM("A")='world'), Row(TRIM("A")='!')]
+
+    Example::
```

Comment:
Maybe we don't need this `Example::` again? Otherwise we should given it a number like `Example 2::`

Reply:
I've seen some other functions with multiple examples structured in this way. Should we change them all then with numbers? I feel a better solution may be to rewrite the doc to guarantee correct example number via a precommit. That seems like a good amount of work though - worth it?

Reply:
No strong opinion. I just saw that `udf.py` has numbered examples.

Reply:
good to know, in that case I'll leave it unnumbered for now - but it's something we should think about for the future. Maybe a guideline somewhere on how to write docs within?